### ISSUE TITLE
Test failures with Git 2.35.0.rc2

### ISSUE THREAD

**Issue by thmo** [mackyle/topgit#22]
The Fedora 36 mass rebuild showed several failures of tg 0.19.13 with Git 2.35.0 (rc2):
```
t5020-update-remote.sh           (Wstat: 0 Tests: 7 Failed: 7)
  Failed tests:  1-7
t5010-update-local.sh            (Wstat: 0 Tests: 8 Failed: 4)
  Failed tests:  1-4
t5030-update-newbranch.sh        (Wstat: 0 Tests: 6 Failed: 6)
  Failed tests:  1-6
t4200-out-of-date.sh             (Wstat: 0 Tests: 59 Failed: 39)
  Failed tests:  1-2, 9, 12-19, 22, 25, 28, 31-47, 49, 52-58
t5060-update-continue.sh         (Wstat: 0 Tests: 5 Failed: 1)
  Failed test:  5
t7000-push.sh                    (Wstat: 0 Tests: 26 Failed: 1)
  Failed test:  1
t8400-export-dag.sh              (Wstat: 0 Tests: 5 Failed: 5)
  Failed tests:  1-5
```

At first glance most of them (if not all) seem to fail with the same pattern:
```
++ exec /usr/bin/git checkout -f stage
BUG: builtin/checkout.c:1098: should be able to skip past 'refs/heads/' in 'refs/top-bases/stage'!
[...]/topgit-0.19.13/t/test-lib-functions.sh: line 1154: 2098311 Aborted
(core dumped) ( { "unset" -f git; } > /dev/null 2>&1 || :; "exec" "$GIT_PATH" "$@" )
```
(from `cd t; ./t8400-export-dag.sh -x -i`).

So, an internal Git bug, I guess?

**Comment by thmo** [comment]
Maybe a related report: https://bugzilla.redhat.com/show_bug.cgi?id=2042920

**Comment by thmo** [comment]
Seems to be fixed in Git 2.35.0.
